inspector/src/components/actionTabs/tabs/propertyGrids/animations/timeline.tsx

```diff
@@ -163,23 +251,125 @@ export class Timeline extends React.Component<
   scrollDragStart(e: React.MouseEvent<HTMLDivElement, MouseEvent>): void;
   scrollDragStart(e: any) {
     e.preventDefault();
-    if ((e.target.class = 'scrollbar') && this._scrollbarHandle.current) {
-      this._scrolling = true;
+    if (e.target.className === 'scrollbar') {
+      if ((e.target.class = 'scrollbar') && this._scrollbarHandle.current) {
+        this._scrolling = true;
+        this._shiftX =
+          e.clientX -
+          this._scrollbarHandle.current.getBoundingClientRect().left;
+        this._scrollbarHandle.current.style.left =
+          e.pageX - this._shiftX + 'px';
+      }
+    }
+
+    if (
+      e.target.className === 'left-grabber' &&
+      this._scrollbarHandle.current
+    ) {
+      this._active = 'leftDraggable';
+      this._shiftX =
+        e.clientX - this._scrollbarHandle.current.getBoundingClientRect().left;
+    }
+
+    if (
+      e.target.className === 'right-grabber' &&
+      this._scrollbarHandle.current
+    ) {
+      this._active = 'rightDraggable';
       this._shiftX =
         e.clientX - this._scrollbarHandle.current.getBoundingClientRect().left;
-      this._scrollbarHandle.current.style.left = e.pageX - this._shiftX + 'px';
     }
   }
 
   scrollDrag(e: React.TouchEvent<HTMLDivElement>): void;
   scrollDrag(e: React.MouseEvent<HTMLDivElement, MouseEvent>): void;
   scrollDrag(e: any) {

```

Comment:
Turned out to be quite a function :-)

Is there a way to separate it to smaller functions just to keep the function a bit smaller? Just if you see a nice and clean way to do that.

Reply:
sure I can. I can split it in three. 1. Handle scrollbar dragging 2. Handle left resize 3.Handle right resize